enumerables.rb

```diff
@@ -0,0 +1,203 @@
+module Enumerable
+  # my_each
+
+  def my_each
+    if block_given?
+      if is_a?(Range)
+        arr = to_a
+        arr.each do |value|
+          yield(value)
+        end
+      else
+        each do |value|
+          yield(value)
+        end
+      end
+      self
+    else
+      to_enum :my_each
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    if block_given?
+      arr = if is_a?(Range)
+              to_a
+            else
+              self
+            end
+      (0..arr.length - 1).each do |index|
+        yield(arr[index], index)
+      end
+      self
+    else
+      to_enum :my_each_with_index
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    new_a = []
+    my_each do |element|
+      new_a << element if yield(element)
+    end
+    new_a
+  end
+
+  # My_all?
+  def my_all?(arg = nil)
+    arr = to_a if self.class == Range
+    arr = self if self.class != Range
+    if block_given?
+      arr.length.times do |i|
+        return false if yield(arr[i]) == false
+      end
+    else
+      puts 'block was not given'
+      puts "self class is #{arr.class}"
+      puts "argument class is #{arg.class}"
+      if !arg.nil?
+        arr.length.times do |i|
+          if arg.class == Integer
+            unless arr[i].to_s.include?(arg.to_s)
+              puts(" array item #{arr[i]} doesn't equal argument #{arg}, rendering false")
+              return false
+            end
+          elsif arg.class == Class
+            unless arr[i].class == arg
+              puts(" array item #{arr[i]} doesn't equal argument #{arg}, rendering false")
+              return false
+            end
+          else
+            unless arr[i].to_s.include?(arg.source)
+              puts(" array item #{arr[i]} doesn't equal argument #{arg.source}, rendering false")
+              return false
+            end
+          end
+        end
+      else
+        arr.length.times do |i|
+          if arr[i] == false || arr[i].nil?
+            puts arr[i]
+            return false
+          end
+        end
+      end
+    end
+    true
+  end
```

Comment:
Your `my_all` produces this error:

```
  2) enumerables #my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
     Failure/Error: expect(numbers.my_all?(Numeric)).to be numbers.all?(Numeric)
     
       expected true
            got false
     # ./spec/enumerables_spec.rb:176:in `block (4 levels) in <top (required)>'
```

Please make sure your `my_all` can handle class names as an argument. 

You may test this by using it thus:

```
   numbers = [1, 2, 3, 4, 5]
# This should return true
   numbers.my_all?(Numeric)
```